frontend/packages/ceph-storage-plugin/src/features.ts

```diff
@@ -1,41 +1,35 @@
 import * as _ from 'lodash';
-import { k8sGet, K8sResourceKind } from '@console/internal/module/k8s';
+import { Dispatch } from 'react-redux';
+import { k8sGet } from '@console/internal/module/k8s';
 import { ClusterServiceVersionModel, SubscriptionModel } from '@console/operator-lifecycle-manager';
-import { handleError, setFlag } from '@console/internal/actions/features';
+import { setFlag } from '@console/internal/actions/features';
 import { FeatureDetector } from '@console/plugin-sdk';
+import { getAnnotations } from '@console/shared/src/selectors/common';
 import { OCSServiceModel } from './models';
 import {
   OCS_INDEPENDENT_CR_NAME,
   CEPH_STORAGE_NAMESPACE,
   OCS_SUPPORT_ANNOTATION,
+  OCS_CONVERGED_CR_NAME,
 } from './constants';
-import { getAnnotations } from '@console/shared/src/selectors/common';
 
 export const OCS_INDEPENDENT_FLAG = 'OCS_INDEPENDENT';
-
-const isIndependent = (data: K8sResourceKind): boolean =>
-  data.spec?.externalStorage?.enable ?? false;
-
-export const detectIndependentMode: FeatureDetector = (dispatch) =>
-  k8sGet(OCSServiceModel, OCS_INDEPENDENT_CR_NAME, CEPH_STORAGE_NAMESPACE).then(
-    (obj: K8sResourceKind) => dispatch(setFlag(OCS_INDEPENDENT_FLAG, isIndependent(obj))),
-    (err) => {
-      err?.response?.status === 404
-        ? dispatch(setFlag(OCS_INDEPENDENT_FLAG, false))
-        : handleError(err, OCS_INDEPENDENT_FLAG, dispatch, detectIndependentMode);
-    },
-  );
+export const OCS_CONVERGED_FLAG = 'OCS_CONVERGED';
+// Used to activate NooBaa dashboard
+export const OCS_FLAG = 'OCS';
+// Todo(bipuladh): Remove this completely in 4.6
+export const CEPH_FLAG = 'CEPH';
 
 /* Key and Value should be same value received in CSV  */
 export const OCS_SUPPORT_FLAGS = {
   SNAPSHOT: 'SNAPSHOT',
   EXTERNAL: 'EXTERNAL',
 };
 
-const handleOCSError = (res, flags, dispatch, cb) => {
+const handleError = (res: any, flags: string[], dispatch: Dispatch, cb: FeatureDetector) => {
```

Comment:
I'd suggest we adapt the existing `handleError` function in `public/actions/features.ts` to support this specific use case:

```ts
const retryFlagDetection = (dispatch: Dispatch, cb) => {
  setTimeout(() => cb(dispatch), 15000);
};

export const handleError = (res, flags: string | string[], dispatch: Dispatch, cb) => {
  const status = res?.response?.status;
  if (_.includes([403, 502], status)) {
    _.castArray(flags).forEach((f) => dispatch(setFlag(f, undefined)));
  }
  if (!_.includes([401, 403, 500], status)) {
    retryFlagDetection(dispatch, cb);
  }
};
```